### PR TITLE
Architecture: Cleanup NodeCreate syscalls

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine-interface/src/api/types/re_node.rs [hunk 1]
```diff
@@ -11,6 +11,28 @@ pub enum ScryptoRENode {
...
+pub enum RENodeType {
+    Bucket,
+    Proof,
+    AuthZoneStack,
+    Worktop,
+    GlobalAccount,
+    GlobalComponent,
+    GlobalResourceManager,
+    GlobalPackage,
+    GlobalEpochManager,
```
Comment (iamyulong) [inline radix-engine-interface/src/api/types/re_node.rs:26]: Feel like we should represent globals as `Global(EntityTypeId)`.

File: radix-engine/src/engine/system_api.rs [hunk 2]
```diff
@@ -71,8 +71,13 @@ pub trait SystemApi {
...
     fn drop_node(&mut self, node_id: RENodeId) -> Result<HeapRENode, RuntimeError>;
 
+    /// Allocates a new node id useable for create_node
+    fn allocate_node_id(&mut self, node_type: RENodeType) -> Result<RENodeId, RuntimeError>;
+
     /// Creates a new RENode
-    fn create_node(&mut self, re_node: RENode) -> Result<RENodeId, RuntimeError>;
+    /// TODO: Remove, replace with lock_substate + get_ref_mut use
+    fn create_node(&mut self, node_id: RENodeId, re_node: RENode)
+        -> Result<RENodeId, RuntimeError>;
```
Comment (iamyulong) [inline radix-engine/src/engine/system_api.rs:80]: Understand you have a TODO there, but is there reason we return `RENodeId`?

Comment (talekhinezh) [reply radix-engine/src/engine/system_api.rs:80]: More to keep interface there once we implement local handles. Would you rather it just be removed?

File: radix-engine-interface/src/api/types/re_node.rs (hunk 1 above)
Comment (talekhinezh) [reply radix-engine-interface/src/api/types/re_node.rs:26]: Not sure if that gives any benefit at the moment? I'd rather avoid complexity until we get to the point of refactoring this area.

Comment (iamyulong) [reply radix-engine-interface/src/api/types/re_node.rs:26]: More from representation accuracy perspective. None of the variants are really a `RENodeType`.

**Pull request merged**